client/post-editor/editor-drawer/copy-post.jsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import Accordion from 'components/accordion';
+import AccordionSection from 'components/accordion/section';
+import Button from 'components/button';
+import Gridicon from 'components/gridicon';
+import PostSelector from 'my-sites/post-selector';
+
+class EditorDrawerCopyPost extends Component {
+
+	static propTypes = {
+		site: PropTypes.object.isRequired,
+		post: PropTypes.object.isRequired,
+		translate: PropTypes.func,
+	};
+
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			selectedPostID: '',
+		};
+	}
+
+	setPostToCopy = post => {
+		this.setState( {
+			selectedPostID: post.ID,
+		} );
+	}
+
+	goToNewDraft = () => {
+		if ( '' !== this.state.selectedPostID ) {
+			page.redirect( `/post/${ this.props.site.slug }/${ this.state.selectedPostID }?copy=true` );
```

Comment:
We'll probably want to strip this query parameter so we don't run the risk of the user refreshing and creating multiple separate copies. The [`?sb` and `?sp` behavior](https://github.com/Automattic/wp-calypso/blob/37d3a88e3732ec131ad518eb0266ed076855e927/client/boot/index.js#L251-L261) could be a reference.